includes/class-wc-payments-payment-method-messaging-element.php

```diff
@@ -65,8 +73,9 @@ public function init(): string {
 			'WCPAY_PRODUCT_DETAILS',
 			'wcpayStripeSiteMessaging',
 			[
-				'price'          => WC_Payments_Utils::prepare_amount( $price, $currency_code ),
+				'productId'      => 'base_product',
 				'currency'       => $currency_code,
+				'productPrices'  => $product_prices,
```

Comment:
Final implementation request: let's move currency into product prices, it looks like:

```
[
    'product-id-here' => [
        'amount'   => ...,
        'currency' => ...,
    ],
    ...
]
```

This change in data structure better fits to the price definition IMO.

Reply:
Correspondingly `currency` after `productId` have to be dropped.

Reply:
Good learning for me in terms of keeping data well-structured, without duplication and clean. Thanks! I did that here and refactored code accordingly  

97f1a29